File: spec/python/test_process_xor4_const.py
import unittest

from process_xor4_const import ProcessXor4Const

class TestProcessXor4Const(unittest.TestCase):
    def test_process_xor4_const(self):
        with ProcessXor4Const.from_file("src/process_xor_4.bin") as r:

            self.assertEqual(r.key, bytearray([0xec, 0xbb, 0xa3, 0x14]))
            self.assertEqual(r.buf, b"foo bar")

File: spec/lua/test_process_xor4_const.lua
local luaunit = require("luaunit")

require("process_xor4_const")

TestProcessXor4Const = {}

function TestProcessXor4Const:test_process_xor4_const()
    local r = ProcessXor4Const:from_file("src/process_xor_4.bin")

    luaunit.assertEquals(r.key, "\xec\xbb\xa3\x14")
    luaunit.assertEquals(r.buf, "foo bar")
end
